### PR TITLE
Making the initial pull request:  DMS_REPLICATION_NOT_PUBLIC

### PR DESCRIPTION
**Pull request by Geeth26**
I confirm these files are made available under CC0 1.0 Universal (https://creativecommons.org/publicdomain/zero/1.0/legalcode)

*Issue #, if available:*

*Description of changes:*

### CODE REVIEW

**Comment by ShrikantDavange** [comment]
Hi,

Please i) add gherkin to the top of code ii) change DEFAULT_RESOURCE_TYPE = 'AWS::::Account' to  AWS::DMS::ReplicationInstance
This is updated gherkin https://github.com/awslabs/aws-config-rules/issues/251

File: python/DMS_REPLICATION_NOT_PUBLIC/DMS_REPLICATION_NOT_PUBLIC_test.py [hunk 1]
```diff
@@ -0,0 +1,204 @@
...
+
+class SampleTest(unittest.TestCase):
+
+    describe_dms_instance_sce_2 = {"ReplicationInstances": [{'ReplicationInstanceIdentifier':'instance1', 'PubliclyAccessible': True}, {'ReplicationInstanceIdentifier':'instance2', 'PubliclyAccessible': True}]}
+    describe_dms_instance_sce_3 = {"ReplicationInstances": [{'ReplicationInstanceIdentifier':'instance1', 'PubliclyAccessible': False}, {'ReplicationInstanceIdentifier':'instance2', 'PubliclyAccessible': False}]}
+    describe_dms_instance_sce_4 = {"ReplicationInstances": [{'ReplicationInstanceIdentifier':'instance1', 'PubliclyAccessible': True}, {'ReplicationInstanceIdentifier':'instance2', 'PubliclyAccessible': False}]}
+
+
+    invoking_event_iam_role_sample = '{"configurationItem":{"relatedEvents":[],"relationships":[],"configuration":{},"tags":{},"configurationItemCaptureTime":"2018-07-02T03:37:52.418Z","awsAccountId":"123456789012","configurationItemStatus":"ResourceDiscovered","resourceType":"AWS::IAM::Role","resourceId":"some-resource-id","resourceName":"some-resource-name","ARN":"some-arn"},"notificationCreationTime":"2018-07-02T23:05:34.445Z","messageType":"ConfigurationItemChangeNotification"}'
```
Comment (jongogogo) [inline python/DMS_REPLICATION_NOT_PUBLIC/DMS_REPLICATION_NOT_PUBLIC_test.py]: remove

File: python/DMS_REPLICATION_NOT_PUBLIC/DMS_REPLICATION_NOT_PUBLIC.py [hunk 2]
```diff
@@ -0,0 +1,405 @@
...
+        #SCENARIO 2 PubliclyAccessible is set to True for the AWS Replication instance
+            evaluations.append(build_evaluation(dms_instance_details['ReplicationInstanceIdentifier'], 'NON_COMPLIANT', event, annotation='This AWS Replication instance has public internet access.'))
+        else:
+            #SCENARIO 2 PubliclyAccessible is set to False for the AWS Replication instance
+            evaluations.append(build_evaluation(dms_instance_details['ReplicationInstanceIdentifier'], 'COMPLIANT', event))
+
+    return evaluations
+
+def evaluate_parameters(rule_parameters):
+    """Evaluate the rule parameters dictionary validity. Raise a ValueError for invalid parameters.
```
Comment (jongogogo) [inline python/DMS_REPLICATION_NOT_PUBLIC/DMS_REPLICATION_NOT_PUBLIC.py]: remove comments

**Comment by jongogogo** [comment]
When is the update planned?

**Comment by ShrikantDavange** [comment]
Hi Sangeeth, can we please have an update

**Comment by ShrikantDavange** [comment]
You can subscribe to this thread to get notification about comments.

**Pull request merged**